### ISSUE TITLE
Add mergeInto() function

### ISSUE THREAD

**Issue by Phillipus** [archimatetool/archi-scripting-plugin#32]
Add new methods:

```java
ArchimateConceptProxy#mergeInto(newElementProxy, deleteOld);
ArchimateConceptProxy#mergeInto(newElementProxy);
```

The "old" element gets deleted and its properties are "merged" (ie. added) to the newElementProxy.
The second method would then simply call `mergeInto(newElementProxy, false)`

Documentation should be appended to newElementProxy's documentation too.

* Add `ArchimateElementProxy#mergeInto(ArchimateElementProxy newElementProxy)` method that uses the passed proxy element instead of creating a new one. Care should be taken so that changes on `newElementProxy` properties are tracked in commands.
* Refactor `ArchimateElementProxy#setType` to create a `newElementProxy` and call `mergeInto()`
* Now this should allow to merge a concept into another.
* It could then be possible to further refactor by creating `DiagramModelComponentProxy#setConcept(ArchimateElementProxy newElementProxy)` using the code inside "Update all diagram objects" section of `ArchimateElementProxy#mergeInto` and then refactor ArchimateElementProxy#mergeInto to use DiagramModelComponentProxy#setConcept.

**Comment by jbsarrodie** [comment]
That's an interresting new pair of methods ;-)

**Comment by Phillipus** [comment]
> That's an interresting new pair of methods ;-)

Yeah, they just came to me now.... ;-)

(we are joking, these are JB's ideas from emails)

**Comment by MarcF88** [comment]
Would this feature provide the long sought after "merge/re-use of elements in two models" functionality (aka Treasure Island of Archimate)? 
Curious for your emailexchange, maybe you could condense them into a blog or something.

**Comment by jbsarrodie** [comment]
>Would this feature provide the long sought after "merge/re-use of elements in two models" functionality (aka Treasure Island of Archimate)?

Not directly, but that's a potential step in this direction.

>Curious for your emailexchange, maybe you could condense them into a blog or something.

Not sure there's enough for a blog post. Basically the goal is to be able to easily and quickly fix a common issue: duplicate elements. This happens frequently when several people maintain a model. So either we become more strict on who can edit the model (and end up with a human SPOF) or we accept that some people will create duplicates and we fix them afterward.

**Comment by Phillipus** [comment]
Implemented in 0.6.0
